src/Symfony/Component/OptionsResolver/OptionConfigurator.php

```diff
@@ -0,0 +1,130 @@
+ */
+
+namespace Symfony\Component\OptionsResolver;
+
+use Symfony\Component\OptionsResolver\Exception\AccessException;
+use Symfony\Component\OptionsResolver\Exception\UndefinedOptionsException;
+
+final class OptionConfigurator
+{
+    /**
+     * The name of the defined option
+     * @var string
+     */
+    private $name;
+
+    /**
+     * @var OptionsResolver
+     */
+    private $resolver;
+
+    public function __construct(string $name, OptionsResolver $resolver)
+    {
+        $this->name = $name;
+        $this->resolver = $resolver;
+        $this->resolver->setDefined($name);
+    }
+
+    /**
+     * Adds allowed types for this option.
+     *
+     * @param string|string[] $types One or more accepted types
+     *
+     * @return $this
+     *
+     * @throws UndefinedOptionsException If the option is undefined
+     * @throws AccessException           If called from a lazy option or normalizer
+     */
+    public function allowedTypes($types): self
+    {
+        $this->resolver->addAllowedTypes($this->name, $types);
+
+        return $this;
+    }
+
+    /**
+     * Sets allowed values for this option.
+     *
+     * @param mixed $values One or more acceptable values/closures
+     *
+     * @return $this
+     *
+     * @throws UndefinedOptionsException If the option is undefined
+     * @throws AccessException           If called from a lazy option or normalizer
+     */
+    public function allowedValues($values): self
+    {
+        $this->resolver->setAllowedValues($this->name, $values);
+
+        return $this;
+    }
+
+    /**
+     * Sets the default value of a given option.
+     *
+     * @param mixed $value The default value of the option
+     *
+     * @return $this
+     *
+     * @throws AccessException If called from a lazy option or normalizer
+     */
+    public function default($value): self
+    {
+        $this->resolver->setDefault($this->name, $value);
+
+        return $this;
+    }
+
+    /**
+     * Marks this option as deprecated.
+     *
+     * @return $this
+     */
+    public function deprecated(): self
+    {
+        $this->resolver->setDeprecated($this->name);
+
+        return $this;
+    }
+
+    /**
+     * Sets the normalizer for this option.
+     *
+     * @param \Closure $normalizer The normalizer
+     *
+     * @return $this
+     *
+     * @throws UndefinedOptionsException If the option is undefined
+     * @throws AccessException           If called from a lazy option or normalizer
+     */
+    public function normalize(\Closure $normalizer): self
+    {
+        $this->resolver->setNormalizer($this->name, $normalizer);
+
+        return $this;
+    }
+
+    /**
+     * Marks this option as required.
+     *
+     * @return $this
+     *
+     * @throws AccessException If called from a lazy option or normalizer
+     */
+    public function required(): self
+    {
+        $this->resolver->setRequired($this->name);
+
+        return $this;
+    }
+}
```

Comment:
I think can be useful to add also `addNormalizer`, `addAllowedValues` and `addAllowedTypes` methods as well

Reply:
I'd also add a `define()` method here to cover the @ro0NL's [idea](https://github.com/symfony/symfony/pull/33848#issuecomment-538438733), e.g:
```php
public function define(string $optionName): self
{
    return $this->resolver->define($optionName);
}
```